torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -222,9 +224,14 @@ Tensor pow_backward_exponent(Tensor grad, const Tensor& self, const Tensor& expo
 }
 
 Tensor pow_backward_exponent(Tensor grad, const Scalar & base, const Tensor& exponent, Tensor result) {
-  auto base_ = base.isComplex() ? base.toComplexDouble() : base.toDouble();
-  auto grad_lambda = [](auto a, auto b) { return (a * std::log(b)).conj(); };
-  if (base_ == 0.0) {
+  auto grad_lambda = [](Tensor a, Scalar b) {
+    return AT_DISPATCH_DOUBLE_COMPLEXDOUBLE(b.type(), "scalar_val", ([&] {
+      scalar_t val = b.to<scalar_t>();
+      return (a * std::log(val)).conj();
```

Comment:
This seems a little goofy. Why not also add a `log()` method on `Scalar`?

Reply:
Should Scalar have more operations defined on it? For example, in `pow_backward`, we would need to define an operator- for Scalar to avoid the dispatch

Reply:
@ezyang added log.
I think it makes sense to support more math operations for `Scalar`. For now, I added `log`, and `equal` in `Scalar.h` but we should add a new file `ScalarMath.{h,cpp}` if we plan to add more ops for scalar